roles/edge-image-builder/tasks/main.yml

```diff
@@ -0,0 +1,77 @@
+---
+- name: Create eib images directory
+  file:
+    path: "{{ eib_path }}/images"
+    state: directory
+
+- name: Copy OS image to eib images directory
+  copy:
+    src: "{{ eib_source_image_path }}/{{ eib_source_image_name }}"
+    dest: "{{ eib_path }}/images/{{ eib_source_image_name }}"
+
+- name: Copy config eib to folder
+  template:
+    src: config-eib.yaml.j2
+    dest: "{{ eib_path }}/config-eib.yaml"
+
+- name: Create the eib custom scripts directory
+  file:
+    path: "{{ eib_path }}/custom/scripts"
+    state: directory
+
+- name: Copy file into the eib custom scripts directory (fix growfs issue)
+  copy:
+    dest: "{{ eib_path }}/custom/scripts/fix-growfs.sh"
+    content: |
+      #!/bin/bash
+      growfs() {
+        mnt="$1"
+        dev="$(findmnt --fstab --target ${mnt} --evaluate --real --output SOURCE --noheadings)"
+        # /dev/sda3 -> /dev/sda, /dev/nvme0n1p3 -> /dev/nvme0n1
+        parent_dev="/dev/$(lsblk --nodeps -rno PKNAME "${dev}")"
+        # Last number in the device name: /dev/nvme0n1p42 -> 42
+        partnum="$(echo "${dev}" | sed 's/^.*[^0-9]\([0-9]\+\)$/\1/')"
+        ret=0
+        growpart "$parent_dev" "$partnum" || ret=$?
+        [ $ret -eq 0 ] || [ $ret -eq 1 ] || exit 1
+        /usr/lib/systemd/systemd-growfs "$mnt"
+      }  
+      growfs /
+
+- name: Git checkout
+  ansible.builtin.git:
+    repo: 'https://github.com/suse-edge/edge-image-builder.git'
+    dest: "{{ working_dir }}/edge-image-builder"
+
+- name: build the EIB container image
+  containers.podman.podman_image:
+    name: eib:dev
+    path: "{{ working_dir }}/edge-image-builder"
+
+- name: Run EIB container to generate the custom image
+  containers.podman.podman_container:
+    name: eib
+    rm: true
+    image: localhost/eib:dev
+    volumes:
+      - "{{ eib_path }}:/eib"
+    detach: false
+    state: started
+    force_restart: no
+    command: "/bin/eib -config-file config-eib.yaml -config-dir /eib -build-dir /eib/_build"
```

Comment:
This seems to be failing with the latest EIB - I wonder if we should pin to a specific commit or start adding release tags to the EIB repo:

```
shardy@localhost:~/metal3-demo-files/eib/_build/build-Jan10_14-46-38> cat raw-build.log
guestfish: no operating system was found on this disk

If using guestfish ‘-i’ option, remove this option and instead
use the commands ‘run’ followed by ‘list-filesystems’.
You can then mount filesystems you want by hand using the
‘mount’ or ‘mount-ro’ command.

If using guestmount ‘-i’, remove this option and choose the
filesystem(s) you want to see by manually adding ‘-m’ option(s).
Use ‘virt-filesystems’ to see what filesystems are available.

If using other virt tools, this disk image won’t work
with these tools.  Use the guestfish equivalent commands
(see the virt tool manual page).
shardy@localhost:~/metal3-demo-files/eib/_build/build-Jan10_14-46-38> cat eib-build.log
2024-01-10T14:46:38.438Z	INFO	combustion/network.go:50	Configuring network component...
2024-01-10T14:46:38.439Z	INFO	combustion/network.go:54	Skipping network component, configuration is not provided
2024-01-10T14:46:38.439Z	INFO	combustion/elemental.go:30	Skipping elemental registration component, configuration is not provided
2024-01-10T14:46:50.521Z	FATAL	eib/main.go:126	An error occurred building the image: running the image modification script: exit status 1
main.main
	/src/cmd/eib/main.go:126
runtime.main
	/usr/lib64/go/1.21/src/runtime/proc.go:267

```

Reply:
@hardys FWIW, I've never seen such error. Is the input image file valid?

Reply:
@atanasdinov I think it's because the downloaded file is xz compressed e.g `SLE-Micro.x86_64-5.5.0-Default-GM.raw.xz` - I incorrectly assumed EIB would consume the file as downloaded but it has to be uncompressed first.

Simple fix but IMO it would be better for EIB to transparently handle this (to reduce the steps necessary for the user, and to reduce storage requirements for the base-image)

Reply:
Oh, I see. Yes, you have to uncompress it first. I'll file a GitHub issue for this - it can be a nice enhancement.